src/Security/Authentication/Cookies/samples/CookieSample/Program.cs

```diff
@@ -1,32 +1,73 @@
-using System.IO;
+using System;
+using System.Linq;
+using System.Security.Claims;
 using System.Threading.Tasks;
-using Microsoft.AspNetCore.Hosting;
-using Microsoft.Extensions.Hosting;
-using Microsoft.Extensions.Logging;
+using Microsoft.AspNetCore.Authentication;
+using Microsoft.AspNetCore.Authentication.Cookies;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Extensions.DependencyInjection;
 
-namespace CookieSample
-{
-    public static class Program
+var builder = WebApplication.CreateBuilder(args);
+
+builder.Services
+    .AddAuthentication(CookieAuthenticationDefaults.AuthenticationScheme)
+    .AddCookie(options =>
     {
-        public static Task Main(string[] args)
+        options.ExpireTimeSpan = TimeSpan.FromSeconds(20);
+        options.Events = new CookieAuthenticationEvents()
         {
-            var host = new HostBuilder()
-                .ConfigureWebHost(webHostBuilder =>
-                {
-                    webHostBuilder
-                        .UseKestrel()
-                        .UseContentRoot(Directory.GetCurrentDirectory())
-                        .UseIISIntegration()
-                        .UseStartup<Startup>();
-                })
-                .ConfigureLogging(factory =>
+            OnCheckSlidingExpiration = context =>
+            {
+                // If 25% expired instead of the default 50%.
+                context.ShouldRenew = context.TimeElapsed > (context.Options.ExpireTimeSpan / 4);
+
+                // Don't renew on API endpoints that use JWT.
+                var authData = context.HttpContext.GetEndpoint()?.Metadata.GetMetadata<IAuthorizeData>();
+                if (authData != null && string.Equals(authData.AuthenticationSchemes, "Bearer", StringComparison.Ordinal))
```

Comment:
You could consider moving this to auth samples instead of manual and adding a test for this scenario (always nicer to have coverage)

https://github.com/dotnet/aspnetcore/blob/main/src/Security/test/AuthSamples.FunctionalTests/CookiesTests.cs

Reply:
I mainly wanted to show @brockallen that there were other patterns than matching request paths.